### PR TITLE
[Dashboard bug] Slice doesn't always show loading icon when loading

### PR DESCRIPTION
**Pull request by graceguo-supercat**
@michellethomas

### CODE REVIEW

File: superset/assets/javascripts/chart/Chart.jsx [hunk 1]
```diff
@@ -146,7 +146,7 @@ class Chart extends React.PureComponent {
   }
 
   render() {
-    const isLoading = this.props.chartStatus === 'loading';
+    const isLoading = [undefined, 'loading'].indexOf(this.props.chartStatus) !== -1;
```
Comment (mistercrunch) [inline superset/assets/javascripts/chart/Chart.jsx]: I feel like this  should either be a `defaultProps` or handled by the parent. @michellethomas what do you think?

Comment (graceguo-supercat) [reply superset/assets/javascripts/chart/Chart.jsx]: this `isLoading` logic is to show <Loading> component when fetching query data. 
I remember you mentioned want to keep `loading` related logic inside `Chart`:
https://github.com/apache/incubator-superset/pull/3581/commits/ab59c4ccc1beeb446d9691f3384ec4e9e1754f6b

Comment (mistercrunch) [reply superset/assets/javascripts/chart/Chart.jsx]: I just meant that we souldn't have and handle an `undefined` state. My impression is that there should be a defined `chartStatus` from the moment the component is created. I'd have to look at the code upstream, but maybe it's a matter of having `loading` set when creating the chart object.

I do think that the load image should be shown inside the Chart component though as it is now.

Comment (graceguo-supercat) [reply superset/assets/javascripts/chart/Chart.jsx]: now i understand. fixed in the initial Chart state.

**Comment by michellethomas** [comment]
lgtm

**Pull request merged**